### DISCUSSION TITLE
How to set tlsv1.3 ciphersuites at context creation time in websocket.

### DISCUSSION THREAD

**Discussion by jinbei09** [openssl/openssl#25942] [Q&A]
Hello~

Our module use websocket to communicate.
Websocket allow setting ciphersuites at context creation time here:
https://github.com/warmcat/libwebsockets/commit/93d4e186c2f45ba4f16cbe1dd107545b606dde89

Now I use nmap command to get the ciphersuites list of my module, and I can get TLSv1.3 like below:
`ddd@framework$ nmap --script ssl-enum-ciphers -p 8642 192.168.31.236
Starting Nmap 7.95 ( https://nmap.org/ ) at 2024-11-13 11:11 CST
Nmap scan report for 192.168.31.236
Host is up (0.0012s latency).

PORT STATE SERVICE
8642/tcp open unknown
| ssl-enum-ciphers:
| TLSv1.2:
| ciphers:
| TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 (ecdh_x25519) - A
| TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 (ecdh_x25519) - A
| TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256 (ecdh_x25519) - A
| compressors:
| NULL
| cipher preference: server
| TLSv1.3:
| ciphers:
| TLS_AKE_WITH_AES_256_GCM_SHA384 (ecdh_x25519) - A
| TLS_AKE_WITH_CHACHA20_POLY1305_SHA256 (ecdh_x25519) - A
| TLS_AKE_WITH_AES_128_GCM_SHA256 (ecdh_x25519) - A
| cipher preference: server
|_ least strength: A

Nmap done: 1 IP address (1 host up) scanned in 0.87 seconds`

But I don't want to use these TLSv1.3 ciphers. I try to set the tls1_3_plus_cipher_list to empty like below:
struct lws_context_creation_info m_info; m_info.tls1_3_plus_cipher_list = "";

But when I use the nmap command to get ciphersuites list, I can still get these TLSv1.3 ciphers.

Is there any way can set the TLSv1.3 ciphers to empty?
I asked in the websocket github repository, but they said that I should ask openssl:
https://github.com/warmcat/libwebsockets/issues/3275

Thanks a lot!!

**Comment by mattcaswell** [comment]
It's unclear from your question: are you trying to disable TLSv1.3 entirely? To have a valid TLSv1.3 configuration you must have at least one ciphersuite.

**Reply by jinbei09** [reply]
Hello~
Yes, I want to disable TLSv1.3 entirely with openssl1.1~
